src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,196 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        // adding the node at the end of the LinkedList
+        Node<T> newNode = new Node<>(value);
+        if (isEmpty()) {
+            head = newNode;
+        } else {
+            newNode.prev = tail;
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        // adding the node into certain place of the list
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            if (isEmpty()) {
+                head = newNode;
+                tail = newNode;
+            } else {
+                newNode.next = head;
+                head = newNode;
+                newNode.next.prev = newNode;
+            }
+        } else if (index == size) {
+            newNode.next = null;
+            newNode.prev = tail;
+            tail = newNode;
+            newNode.prev.next = newNode;
+        } else {
+            // adding the node in the middle
+            try {
+                Node finderNode = head;
+                for (int i = 0; i < index; i++) {
+                    finderNode = finderNode.next;
+                }
+                newNode.next = finderNode;
+                newNode.prev = finderNode.prev;
+                finderNode.prev = newNode;
+                newNode.next.prev = newNode;
+                newNode.prev.next = newNode;
+            } catch (NullPointerException e) {
+                throw new IndexOutOfBoundsException("Exception in thread \"main\" "
+                        + "java.lang.IndexOutOfBoundsException");
+            }
+
+        }
+        size++;
```

Comment:
You can reuse method add() for adding in the end.
Create method for getting element by index and reuse everywhere you need